Invalid generated C code when initializing aliased type

### Describe the bug

This is a weird one. I ran into a case where the compiler generates invalid C code. I am creating a type system for a toy interpreter and I have a module that is defining some value types using type aliases that all implement a simple `Value` interface.

### Reproduction Steps

I distilled it down into a simple module with a test:
```
module value

pub interface Value {
	str() string
}

pub type List = []Value
pub fn (x List) str() string { return x.str() }
```

test:
```
module value

fn test_list() {
	mut list := make_list()!
	assert (list as List).len == 0
}

fn make_list() !Value {
	mut list := List{}
	return list
}
```

### Expected Behavior

Test should pass, `list` should be initialized with zero length

### Current Behavior

This results in a compilation error in the C compiler:

```
cc: C:/Users/Me/AppData/Local/Temp/v_0/tsession_2298_01KCD3T3RC207TSYGBGQ72R8R0/base_test.exe.01KCD3T508YKSKB84NC4K8XYJM.tmp.c:7993: error: ';' expected (got "}")
```

### Possible Solution

Seems like the `HEAP` macro(?) is misbehaving here?

### Additional Information/Context

The generated C code in question looks like this:

```
VV_LOC _result_value__Value value__make_list(void) {
	value__List *list = HEAP(value__List, ({0})); // <-- CRASHES HERE
	_result_value__Value _t1 = {0};
	builtin___result_ok(&(value__Value[]) { I_value__List_to_Interface_value__Value(&(*(list))) }, (_result*)(&_t1), sizeof(value__Value));
	return _t1;
}
```

In order to repro this I have to wrap the `List` initialization in a separate function that returns the `Value` interface. If I return `List` directly, as below, the code compiles and works.
```
module value

fn test_list() {
	mut list := make_list()
	assert list.len == 0
}

fn make_list() List {
	mut list := List{}
	return list
}
```

### V version

0.4.12

### Environment details (OS name and version, etc.)

```
|V full version      |V 0.4.12 55436caa9ae9c179639a4b238fc358ee7d768645.c15523b
|:-------------------|:-------------------
|OS                  |windows, Microsoft Windows 11 Pro 26200 64-bit
|Processor           |28 cpus, 64bit, little endian, 13th Gen Intel(R) Core(TM) i7-13850HX
|Memory              |15.31GB/31.61GB
|                    |
|V executable        |C:\Program Files\v_windows\v\v.exe
|V last modified time|2025-12-14 00:22:30
|                    |
|V home dir          |contains spaces, value: C:\Program Files\v_windows\v
|VMODULES            |contains spaces, value: C:\Users\Casey Duncan\.vmodules
|VTMP                |contains spaces, value: C:\Users\Casey Duncan\AppData\Local\Temp\v_0
|Current working dir |contains spaces, value: C:\Users\Casey Duncan\projects\vebol
|                    |
|Git version         |git version 2.52.0.windows.1
|V git status        |weekly.2025.50-36-gc15523b2
|.git/config present |true
|                    |
|cc version          |N/A
|gcc version         |N/A
|clang version       |N/A
|msvc version        |N/A
|tcc version         |tcc version 0.9.27 (x86_64 Windows)
|tcc git status      |N/A
|emcc version        |N/A
|glibc version       |ldd (cygwin) 3.6.5
```
> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/struct.v
+++ b/vlib/v/gen/c/struct.v
@@ -52,7 +52,7 @@ fn (mut g Gen) struct_init(node ast.StructInit) {
 			g.write(g.type_default_sumtype(unwrapped_typ, sym))
 		}
 		return
-	} else if sym.kind == .map {
+	} else if sym.kind == .map || (sym.kind == .array && node.init_fields.len == 0) {
 		g.write(g.type_default(unwrapped_typ))
 		return
 	}

--- a/vlib/v/tests/aliases/modules/value/value.v
+++ b/vlib/v/tests/aliases/modules/value/value.v
@@ -1,0 +1,11 @@
+module value
+
+pub interface Value {
+	str() string
+}
+
+pub type List = []Value
+
+pub fn (x List) str() string {
+	return x.str()
+}

--- a/vlib/v/tests/aliases/modules/value/value_test.v
+++ b/vlib/v/tests/aliases/modules/value/value_test.v
@@ -1,0 +1,11 @@
+module value
+
+fn test_list() {
+	mut list := make_list()!
+	assert (list as List).len == 0
+}
+
+fn make_list() !Value {
+	mut list := List{}
+	return list
+}

